fix(DB/Quest): Aces High! quest issues

blinkysc opened the pull request:
[!] Please Read! This PR comes from AI with MCP

During investigation of the Aces High! quest issues, a core bug was discovered where flying vehicles using `SPELL_AURA_FLY` or `MOVEMENTFLAG_CAN_FLY` were not granting parachutes on dismount. The original code only checked `IsFlying()` which requires `MOVEMENTFLAG_FLYING | MOVEMENTFLAG_DISABLE_GRAVITY` - flags that creature vehicles typically don't have. This fix expands the check to include all flying vehicle types.

## Changes Proposed:
This PR proposes changes to:
-  [x] Core (units, players, creatures, game systems).
-  [ ] Scripts (bosses, spell scripts, creature scripts).
-  [x] Database (SAI, creatures, etc).

## Issues Addressed:
- Closes https://github.com/azerothcore/azerothcore-wotlk/issues/23834
- Closes https://github.com/chromiecraft/chromiecraft/issues/8561

## SOURCE:
The changes have been validated through:
- [ ] Live research (checked on live servers, e.g Classic WotLK, Retail, etc.)
- [ ] Sniffs (remember to share them with the open source community!)
- [x] Video evidence, knowledge databases or other public sources (e.g forums, Wowhead, etc.)
- [ ] The changes promoted by this pull request come partially or entirely from another project (cherry-pick).

## Tests Performed:
This PR has been:
- [x] Tested in-game by the author.
- [ ] Tested in-game by other community members/someone else other than the author/has been live on production servers.
- [ ] This pull request requires further testing and may have edge cases to be tested.

## How to Test the Changes:
- [x] This pull request can be tested by following the reproduction steps provided in the linked issue
- [ ] This pull request requires further testing.

**Aces High! (Quest 13413):**
1. `.go creature id 32548` - Go to Corastrasza
2. `.quest add 13413` - Add "Aces High!" quest
3. Talk to Corastrasza and request a Skytalon
4. Verify **Blazing Speed** is on button 6 of vehicle bar
5. Engage Scalesworn Elite - verify they cast **Ice Shard** and **Arcane Surge**
6. Fly high and dismount - verify **Parachute** buff is applied

**Vile Like Fire! (Quest 13071)** - Also fixed by the parachute change:
1. `.go creature id 30272` - Go to Njorndar Proto-Drake
2. `.quest add 13071` - Add "Vile Like Fire!" quest
3. Right-click the proto-drake to mount
4. Fly high and dismount - verify **Parachute** buff is applied

**Vehicles affected by parachute fix (spell 57403 - Flight aura):**
```sql
SELECT ct.entry, ct.name, cts.Spell 
FROM creature_template ct 
JOIN creature_template_spell cts ON ct.entry = cts.CreatureID 
WHERE ct.VehicleId > 0 AND cts.Spell = 57403;
```
| entry | name | Spell |
|-------|------|-------|
| 30161 | Wyrmrest Skytalon | 57403 |
| 30564 | Njorndar Proto-Drake | 57403 |
| 31752 | Wyrmrest Skytalon (1) | 57403 |
| 32535 | Wyrmrest Skytalon | 57403 |

Note: 30161 and 31752 are Eye of Eternity instance vehicles (excluded by `!GetInstanceId()` check).

## Known Issues and TODO List:
- [x] None

blinkysc committed:
fix(DB/Quest): Aces High! quest issues

- Add missing Blazing Speed ability to Wyrmrest Skytalon (32535)
- Add Ice Shard and Arcane Surge spells to Scalesworn Elite (32534)
- Fix parachute not applying when dismounting flying vehicles with SPELL_AURA_FLY

Closes https://github.com/azerothcore/azerothcore-wotlk/issues/23834

blinkysc committed:
fix: auto mount on cast

blinkysc commented:
> 1. Revivify still consumes combo points

Client has SPELL_ATTR1_FINISHING_MOVE_DAMAGE for this spell so should be intended. 
https://www.youtube.com/watch?v=zNviZrJ9er8
https://www.youtube.com/watch?v=1OifUBROjeY

> Summoning a Skytalon doesn't put you into it like it should (similar to the drakes in the Occulus)

Addressed with recent commit and should be ready now

rorshan commented:
>Addressed with recent commit and should be ready now

Yep after testing, this works now.

After looking at more comments on wowhead
> "This button says it awards a combo point but it doesn't - it actually CLEARS all combo points when you click on it." -- actually -- it DOES give combo points but on YOURSELF. You can see this if you have self-cast enabled, spam it 5x with no target, then target yourself. The reason it clears the combo points on your Target is that it starts building points on a new target (i.e. yourself). Not entirely sure how to optimally use this but it's not actually lying.

Indeed when I cast revivify i see combo points on the Skytalon, which clears any combo points on other targets. Not sure why it's flagged as a finishing move then.

blinkysc committed:
Update data/sql/updates/pending_db_world/rev_1764209012421370939.sql

Co-authored-by: Gultask <100873791+Gultask@users.noreply.github.com>

blinkysc commented on data/sql/updates/pending_db_world/rev_1764209012421370939.sql:
```diff
@@ -0,0 +1,24 @@
+DELETE FROM `smart_scripts` WHERE `entryorguid` = 32534 AND `source_type` = 0;
+INSERT INTO `smart_scripts` (`entryorguid`, `source_type`, `id`, `link`, `event_type`, `event_phase_mask`, `event_chance`, `event_flags`, `event_param1`, `event_param2`, `event_param3`, `event_param4`, `event_param5`, `event_param6`, `action_type`, `action_param1`, `action_param2`, `action_param3`, `action_param4`, `action_param5`, `action_param6`, `target_type`, `target_param1`, `target_param2`, ...
+(32534, 0, 0, 0, 0, 0, 100, 0, 1000, 2000, 2000, 3000, 0, 0, 11, 61269, 0, 0, 0, 0, 0, 2, 0, 0, 0, 0, 0, 0, 0, 0, 'Scalesworn Elite - In Combat - Cast Ice Shard'),
+(32534, 0, 1, 0, 0, 0, 100, 0, 8000, 12000, 15000, 20000, 0, 0, 11, 61272, 0, 0, 0, 0, 0, 2, 0, 0, 0, 0, 0, 0, 0, 0, 'Scalesworn Elite - In Combat - Cast Arcane Surge');
```
Probably right to have castFlag 64

blinkysc committed:
fix: castFlag 64

blinkysc committed:
Merge branch 'master' into fix-aces-high-quest

blinkysc committed:
fix(DB/Quest): Remove Flight aura (57403) from Skytalon action bar per sniff

Sniff from build 46368 shows 57403 is not on the vehicle action bar for
Wyrmrest Skytalon (32535). Flight is handled via spell 60534 in cooldowns.

Co-Authored-By: bench <heyitsbench@users.noreply.github.com>

blinkysc committed:
fix(DB/Quest): Update Skytalon spell VerifiedBuild to match sniff 46368

Co-Authored-By: bench <heyitsbench@users.noreply.github.com>

sogladev committed:
Merge branch 'master' into fix-aces-high-quest

blinkysc committed:
Update data/sql/updates/pending_db_world/rev_1764209012421370939.sql

Co-authored-by: sogladev <sogladev@gmail.com>

blinkysc committed:
Merge remote-tracking branch 'origin/master' into fix-aces-high-quest

# Conflicts:
#	src/server/game/Entities/Vehicle/Vehicle.cpp

blinkysc committed:
Merge remote-tracking branch 'origin/fix-aces-high-quest' into fix-aces-high-quest
